src/test/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/SyncAndMigrateResourceIntTest.kt

```diff
@@ -1020,4 +1050,95 @@ class SyncAndMigrateResourceIntTest : SqsIntegrationTestBase() {
       }
     }
   }
+
+  @DisplayName("GET /sync/id/{id}")
+  @Nested
+  inner class SyncNomisToLocation {
+
+    @Nested
+    inner class Security {
+      @Test
+      fun `access forbidden when no authority`() {
+        webTestClient.get().uri("/sync/id/${UUID.randomUUID()}")
+          .exchange()
+          .expectStatus().isUnauthorized
+      }
+
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.get().uri("/sync/id/${UUID.randomUUID()}")
+          .headers(setAuthorisation(roles = listOf()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.get().uri("/sync/id/${UUID.randomUUID()}")
+          .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+    }
+
+    @Nested
+    inner class Validation {
+      @Test
+      fun `access client error bad data`() {
+        webTestClient.get().uri("/sync/id/-23rf-$$££@")
+          .headers(setAuthorisation(roles = listOf("ROLE_SYNC_LOCATIONS"), scopes = listOf("write")))
+          .exchange()
+          .expectStatus().is4xxClientError
+      }
+
+      @Test
+      fun `handles location not found`() {
+        webTestClient.get().uri("/sync/id/${UUID.randomUUID()}")
+          .headers(setAuthorisation(roles = listOf("ROLE_SYNC_LOCATIONS"), scopes = listOf("write")))
+          .exchange()
+          .expectStatus().is4xxClientError
+      }
+
+      @Test
+      fun `cannot sync an existing location`() {
+        webTestClient.get().uri("/sync/id/${wingB.id}")
+          .headers(setAuthorisation(roles = listOf("ROLE_SYNC_LOCATIONS"), scopes = listOf("write")))
+          .exchange()
+          .expectStatus().is4xxClientError
+      }
+    }
+
+    @Nested
+    inner class HappyPath {
```

Comment:
Not sure why you are mocking out repository and service?  Why not call the service as normal as there is data there for cell1 and it will be returned in Legacy format.  Also there is a test for /sync/id/ already but happy to add more.

Reply:
@Mjwillis 
1. The idea is touching this method SyncService.**getLegacyLocation**.
2. I need to get _legacylocation_object returned from this service SyncSerivce.getlegacylocation    
	-> I want to believe that I should beable to add this to CommonDataTestBase or add it in the setup()
	else
	I would leave this Mock as it PASS.

Reply:
@Mjwillis 
I have amended/corrected using SynService directly as it suggested (it is better option)
I have deleted repository as it is not needed.
It has increased Positive Hit for **fun getLegacyLocation**